### PR TITLE
filterx: OTel <-> JSON conversion

### PR DESCRIPTION
**Pull request by alltilla**
Now supports:

```
    otel_kvl = otel_kvlist({"foo": 42});
    otel_arr = otel_array([1, 2]);

    js = json();

    js.otel_kvl = otel_kvl;
    istype(js.otel_kvl, "json_object");
    js.otel_kvl += {"bar": 1337};

    js.otel_arr = otel_arr;
    istype(js.otel_arr, "json_array");
    js.otel_arr += [3, 4];
```
and
```
    js = json({"foo": 42});
    js_arr = json_array([1, 2]);

    otel_kvl = otel_kvlist();

    otel_kvl.js = js;
    istype(otel_kvl.js, "otel_kvlist");
    otel_kvl.js += {"bar": 1337};

    otel_kvl.js_arr = js_arr;
    istype(otel_kvl.js_arr, "otel_array");
    otel_kvl.js_arr += [3, 4];
```

### CODE REVIEW

File: lib/filterx/filterx-object.h [hunk 1]
```diff
@@ -173,10 +173,10 @@ filterx_object_clone(FilterXObject *self)
 }
 
 static inline gboolean
-filterx_object_map_to_json(FilterXObject *self, struct json_object **object)
+filterx_object_map_to_json(FilterXObject *self, struct json_object **object, FilterXObject **assoc_object)
 {
   if (self->type->map_to_json)
-    return self->type->map_to_json(self, object);
+    return self->type->map_to_json(self, object, assoc_object);
```
Comment (bazsi) [inline lib/filterx/filterx-object.h:179]: I was wondering if we could set *assoc_object to self right here if the implementation did not do it.

It seems fragile to have to initialize that everywhere.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

File: lib/filterx/filterx-object.h (hunk 1 above)
Comment (bazsi) [reply lib/filterx/filterx-object.h]: I added a followup patch to do just that and with that a lot of extra *assoc_object initializations are gone.

**Review by bazsi — APPROVED** [review]
Merging this with my proposed simplification of the interface. We can always back it out if you disagree.

**Pull request merged**